site/docs/hosting/fly.md

```diff
@@ -0,0 +1,435 @@
+
+<CodeGroup>
+<CodeGroupItem title="Deno (Webhooks)" Active>
+
+```toml{7,11,12}
+# fly.toml
+app = "grammy"
+kill_signal = "SIGINT"
+kill_timeout = 5
+
+[processes]
+  app = "run --allow-net ./app.ts"
+
+[[services]]
+  http_checks = []
+  internal_port = 8000
+  processes = ["app"]
+  protocol = "tcp"
+  script_checks = []
+  [services.concurrency]
+    hard_limit = 25
+    soft_limit = 20
+    type = "connections"
+
+  [[services.ports]]
+    force_https = true
+    handlers = ["http"]
+    port = 80
+
+  [[services.ports]]
+    handlers = ["tls", "http"]
+    port = 443
+
+  [[services.tcp_checks]]
+    grace_period = "1s"
+    interval = "15s"
+    restart_limit = 0
+    timeout = "2s"
+```
+
+</CodeGroupItem>
+<CodeGroupItem title="Deno (Long polling)" Active>
+
+```toml{7}
+# fly.toml
+app = "grammy"
+kill_signal = "SIGINT"
+kill_timeout = 5
+
+[processes]
+  app = "run --allow-net ./app.ts"
+
+# Simply omitting the whole [[services]] section 
+# since we are not listening to HTTP
+```
+
+</CodeGroupItem>
+<CodeGroupItem title="Node.js (Webhooks)" Active>
+
+```toml{7,11,18,19}
+# fly.toml
+app = "grammy"
+kill_signal = "SIGINT"
+kill_timeout = 5
+
+[processes]
+  app = "node ./build/app.js"
+
+# Adjust the NODE_ENV environment to suppress the warning
+[build.args]
+  NODE_ENV = "production"
+
+[build]
+  builder = "heroku/buildpacks:20"
+
+[[services]]
+  http_checks = []
+  internal_port = 8000
+  processes = ["app"]
+  protocol = "tcp"
+  script_checks = []
+  [services.concurrency]
+    hard_limit = 25
+    soft_limit = 20
+    type = "connections"
+
+  [[services.ports]]
+    force_https = true
+    handlers = ["http"]
+    port = 80
+
+  [[services.ports]]
+    handlers = ["tls", "http"]
+    port = 443
+
+  [[services.tcp_checks]]
+    grace_period = "1s"
+    interval = "15s"
+    restart_limit = 0
+    timeout = "2s"
+```
+
+</CodeGroupItem>
+<CodeGroupItem title="Node.js (Long polling)" Active>
+
+```toml{7,11,22,23}
+# fly.toml
+app = "grammy"
+kill_signal = "SIGINT"
+kill_timeout = 5
+
+[processes]
+  app = "node ./build/app.js"
+
+# Adjust the NODE_ENV environment to suppress the warning
+[build.args]
+  NODE_ENV = "production"
+
+[build]
+  builder = "heroku/buildpacks:20"
+
+# Simply omitting the whole [[services]] section 
+# since we are not listening to HTTP
+```
+
+</CodeGroupItem>
+</CodeGroup>
+
+5. Run `flyctl deploy` to deploy your code.
+
+### Method 2: With GitHub Actions
+
+The main advantage of following method is that Fly will watch for changes in your repository which includes your bot code, and it will deploy new versions automatically.
+Visit <https://fly.io/docs/app-guides/continuous-deployment-with-github-actions> for the detailed instructions.
+
+1. Install [flyctl](https://fly.io/docs/hands-on/install-flyctl).
+2. Create a repository on GitHub, it can be either private or public.
+3. Get a Fly API token with `flyctl auth token`.
+4. Go to your newly created repository on GitHub and select "Settings".
```

Comment:
I think English is weird here and requires this, am I wrong?
```suggestion
4. Go to your newly created repository on GitHub and select "Settings."
```

Reply:
It sure looks weird. We can also just omit the quotes

Reply:
What do you think about swapping 2nd and 3rd steps.

2. Get a Fly API token by running `flyctl auth token`.
3. Create a repository on GitHub, it can be either private or public.
4. Go to repository settings.

Or merge 3 & 4.

Reply:
> It sure looks weird.

We can also just ignore this thing, and leave everything as-is. I doubt anyone will complain.

> swapping 2nd and 3rd steps

Great idea!

Reply:
Agree with swapping. For merging, i think no.

3. Create a repository
4. Go to settings, choose secrets, and add api token

Looks better in two steps for me

Reply:
That's better 👌

Reply:
Oh, these were from here: https://fly.io/docs/app-guides/continuous-deployment-with-github-actions/#speed-run-your-way-to-continuous-deployment.

"Go to your newly created repository on GitHub and select Settings." works, but I liked Loskir's suggestion.